### PR TITLE
[MOD-110][Fix] Allow filtering preprints by the node's is_public

### PR DESCRIPTION
**Pull request by aaxelb**
## Purpose
Allow excluding preprints with private nodes from the reviews UI.

## Changes
Allow filtering with `filter[node_is_public]=true` on all preprint list endpoints.

Exclude actions on preprints with private nodes from `/users/me/actions/`

## Side effects
None

## Ticket
https://openscience.atlassian.net/browse/MOD-110

### CODE REVIEW

File: api_tests/reviews/mixins/filter_mixins.py [hunk 1]
```diff
@@ -67,7 +68,7 @@ def allowed_providers(self, providers):
     @pytest.fixture()
     def expected_actions(self, all_actions, allowed_providers):
         provider_ids = set([p.id for p in allowed_providers])
-        return [l for l in all_actions if l.target.provider_id in provider_ids]
+        return [a for a in all_actions if a.target.provider_id in provider_ids]
```
Comment (laurenbarker) [inline api_tests/reviews/mixins/filter_mixins.py]: Would this be clearer if `a` --> `action`?

File: api_tests/users/views/test_user_actions.py [hunk 2]
```diff
@@ -21,6 +21,14 @@ class TestActionFilters(ActionFilterMixin):
     def url(self):
         return '/{}users/me/actions/'.format(API_BASE)
 
+    @pytest.fixture()
+    def expected_actions(self, all_actions, allowed_providers):
+        expected = super(TestActionFilters, self).expected_actions(all_actions, allowed_providers)
+        node = expected[0].target.node
+        node.is_public = False
+        node.save()
+        return [a for a in expected if a.target.node.is_public]
```
Comment (laurenbarker) [inline api_tests/users/views/test_user_actions.py]: Same here, `a` --> `action`

File: api_tests/reviews/mixins/filter_mixins.py [hunk 3]
```diff
@@ -55,7 +56,7 @@ def providers(self):
     def all_actions(self, providers):
         actions = []
         for p in providers:
```
Comment (laurenbarker) [inline api_tests/reviews/mixins/filter_mixins.py]: I'm not a huge fan of one letter variables but ultimately it's up to you.

File: api_tests/users/views/test_user_actions.py [hunk 4]
```diff
@@ -21,6 +21,14 @@ class TestActionFilters(ActionFilterMixin):
...
+
     def test_no_permission(self, app, url, expected_actions):
```
Comment (laurenbarker) [inline api_tests/users/views/test_user_actions.py]: What was `expected_actions` before? (I can't seem to figure out where is came from)

Comment (laurenbarker) [inline api_tests/users/views/test_user_actions.py]: Where did `expected_actions` come from before if it wasn't defined in this file or imported? I feel like I'm missing something...

File: api_tests/reviews/mixins/filter_mixins.py (hunk 1 above)
Comment (aaxelb) [reply api_tests/reviews/mixins/filter_mixins.py]: I think I'm good with one-letter names in list comprehensions, but I agree with you everywhere else.

File: api_tests/users/views/test_user_actions.py (hunk 4 above)
Comment (aaxelb) [reply api_tests/users/views/test_user_actions.py]: It's a pytest fixture [defined on the parent class, `ActionFilterMixin`](https://github.com/aaxelb/osf.io/blob/f549e490357e0b5052e75408acbf7e03eb851aed/api_tests/reviews/mixins/filter_mixins.py#L69). It's the subset of actions that should be visible to the user in the api. The overriding method calls `super()` and filters it further to exclude actions on preprints with private nodes.

File: api_tests/users/views/test_user_actions.py (hunk 2 above)
Comment (aaxelb) [reply api_tests/users/views/test_user_actions.py]: I changed `expected` to `actions` to hopefully make it clearer?

File: api_tests/reviews/mixins/filter_mixins.py (hunk 3 above)
Comment (aaxelb) [reply api_tests/reviews/mixins/filter_mixins.py]: I agree here.

**Pull request merged**